### PR TITLE
chore: Fix typos in 'compiler' (batch 1)

### PR DESCRIPTION
**Pull request by Sajjon**
Batch 1/3: Fixes typos in `compiler`

(See [issue](https://github.com/rust-lang/rust/issues/129874) tracking all PRs with typos fixes)

### CODE REVIEW

File: compiler/rustc_codegen_gcc/src/lib.rs [hunk 1]
```diff
@@ -357,7 +357,7 @@ impl Deref for SyncContext {
 
 unsafe impl Send for SyncContext {}
 // FIXME(antoyo): that shouldn't be Sync. Parallel compilation is currently disabled with "-Zno-parallel-llvm".
-// TODO: disable it here by returing false in CodegenBackend::supports_parallel().
+// TODO: disable it here by returning false in CodegenBackend::supports_parallel().
```
Comment (compiler-errors) [inline compiler/rustc_codegen_gcc/src/lib.rs:360]: rustc_codegen_gcc changes should be reverted and put in their own pr against that repository

**Review by compiler-errors — APPROVED** [review]

File: compiler/rustc_codegen_gcc/src/lib.rs (hunk 1 above)
Comment (Sajjon) [reply compiler/rustc_codegen_gcc/src/lib.rs]: Reverted

**Review by biabbas — APPROVED** [review]

**Review by biabbas — APPROVED** [review]

**Review by jieyouxu — APPROVED** [review]
Thanks!
